edit-post/api/more-menu-item.js

```diff
@@ -0,0 +1,91 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { isString } from 'util';
+import { validateNamespacedId } from '../../utils/plugins';
+import { activateSidebar } from './sidebar';
+
+const menuItems = {};
+
+/**
+ * Registers a plugin under the more menu.
+ *
+ * @param {string}   menuItemId        The unique identifier of the plugin. Should be in
+ *                                     `[namespace]/[name]` format.
+ * @param {Object}   settings          The settings for this menu item.
+ * @param {string}   settings.title    The name to show in the settings menu.
+ * @param {func}     settings.target   The registered plugin that should be activated.
+ * @param {string}   [settings.icon]   SVG Icon url.
+ *
+ * @return {Object} The final sidebar settings object.
+ */
+export function registerMoreMenuItem( menuItemId, settings ) {
+	settings = {
+		menuItemId,
+		...settings,
+	};
+
+	settings = applyFilters( 'editor.registerMoreMenuItem', settings, menuItemId );
+
+	if ( ! validateNamespacedId( menuItemId ) ) {
```

Comment:
All this common validation makes me wonder at what point we should consider reaching for a full-fledge solution built for this type of thing, such as [Yup](https://github.com/jquense/yup) ([Joi](https://github.com/hapijs/joi)) or [JSON Schema](http://json-schema.org/).